pkg/controller/hostpathprovisioner/daemonset.go

```diff
@@ -176,15 +174,30 @@ func createDaemonSetObject(cr *hostpathprovisionerv1.HostPathProvisioner, reqLog
 									Name: "NODE_NAME",
 									ValueFrom: &corev1.EnvVarSource{
 										FieldRef: &corev1.ObjectFieldSelector{
-											APIVersion: "v1",
-											FieldPath:  "spec.nodeName",
+											FieldPath: "spec.nodeName",
```

Comment:
We probably want to keep the APIVersion field.

Reply:
Done